pipenv/environments.py

```diff
@@ -81,7 +81,15 @@ def normalize_pipfile_path(p):
 os.environ.pop("__PYVENV_LAUNCHER__", None)
 # Internal, to tell whether the command line session is interactive.
 SESSION_IS_INTERACTIVE = _isatty(sys.stdout)
-PIPENV_IS_CI = env_to_bool(os.environ.get("CI") or os.environ.get("TF_BUILD") or False)
+
+# TF_BUILD indicates to Azure pipelines it is a build step
+PIPENV_IS_CI = os.environ.get("CI") or os.environ.get("TF_BUILD")
+try:
+    PIPENV_IS_CI = env_to_bool(PIPENV_IS_CI)
```

Comment:
It should suffice to use `os.getenv` instead of `env_to_bool` if you just want to check it the variable was set.